plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -61,11 +66,58 @@ protected void setup(Binder binder)
                 .setDefault()
                 .toInstance(SIMPLE_USER_NAME_PROVIDER);
 
+        install(conditionalModule(
+                ThriftHttpMetastoreConfig.class,
+                config -> config.getAuthenticationMode()
+                        .filter(isEqual(BEARER))
+                        .isPresent(),
+                conditionalBinder -> configBinder(conditionalBinder).bindConfig(ThriftHttpMetastoreConfig.class)));
+
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
 
+        validateThriftHttpConfiguration();
         install(new ThriftMetastoreAuthenticationModule());
     }
 
+    private void validateThriftHttpConfiguration()
+    {
+        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
+        ThriftMetastoreConfig metastoreConfig = buildConfigObject(ThriftMetastoreConfig.class);
+        ThriftHttpMetastoreConfig httpMetastoreConfig = buildConfigObject(ThriftHttpMetastoreConfig.class);
+
+        if (!staticMetastoreConfig.getMetastoreUris().isEmpty()) {
+            boolean hasHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                    .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()));
+            boolean hasHttpsMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                    .anyMatch(uri -> "https".equalsIgnoreCase(uri.getScheme()));
+
+            if (hasHttpMetastore || hasHttpsMetastore) {
+                if (hasHttpMetastore && hasHttpsMetastore) {
+                    throw new IllegalStateException("'hive.metastore.uri' cannot contain both http and https URI schemes");
+                }
```

Comment:
We could move this validation to StaticMetastoreConfig and then deny other combinations (e.g. thrift vs http(s)) as well.